### PR TITLE
Add Course Engagement by Modality plot

### LINKED ISSUE #13: Course Engagement Plot
Add https://github.com/fhdsl/itn-metric/blob/main/last_plot.Rmd

### PR DESCRIPTION
**Pull request by howardbaik**
#### What changes are being implemented in this Pull Request?

I added one last plot from @carriewright11 using the code from https://github.com/fhdsl/itn-metric/blob/main/last_plot.Rmd

#### What was your approach?

Added plot and corresponding caption

#### What GitHub issue does your pull request address?

https://github.com/FredHutch/itn-dashboard/issues/13

### Tell potential reviewers what kind of feedback you are soliciting.

Preview:

Screenshot 2024-03-26 at 11 24 02 AM

### CODE REVIEW

File: app.R [hunk 1]
```diff
@@ -404,12 +433,30 @@ server <- function(input, output) {
...
+      facet_wrap(~modality, nrow = 3, scales = "free_y")  + 
+      scale_fill_manual(values=cbPalette) + 
+      theme_classic() + 
+      theme(axis.text.x = element_text(angle = 45, hjust=1),
+            text = element_text(size = 17, family = "Arial")) + 
+      labs(x = NULL, y = "Number of Learners")
+  })
+  # Caption
+  output$engagement_by_modality_caption <- renderText({
+    "The number of learners for each modality (website = bookdown website) for each course."
```
Comment (cansavvy) [inline app.R]:
```suggestion
    "The number of learners for each modality for each course."
```

I think most folks don't know what we mean by bookdown and not sure it adds clarity. I could be wrong though so happy to figure out a better way to add this note.

File: app.R [hunk 2]
```diff
@@ -261,6 +268,28 @@ server <- function(input, output) {
...
+                                   googlesheets4::read_sheet,
+                                   sheet = "engagement overall") 
+  
+  
+  course_processed <- reactive({
+    course_raw() %>% 
+      pivot_longer(cols = contains("count"), names_to = "modality", values_to = "number_of_learners") %>%
+      mutate(course_name = factor(course_name)) %>%
+      separate(modality, sep = "_", into = c("modality", "meh")) %>% 
+      mutate(modality = factor(modality, levels = c("website", "leanpub", "coursera")))%>%
```
Comment (cansavvy) [inline app.R]:
```suggestion
      mutate(modality = factor(modality, levels = c("website", "leanpub", "coursera")))%>%
```

Can we have Leanpub and Coursera capitalized? Not sure where we change this.

**Review by cansavvy — APPROVED** [review]
This looks great! Only minor comments! Thanks @howardbaek

File: app.R (hunk 1 above)
Comment (howardbaik) [reply app.R]: I deleted the `website = bookdown website`. I admit, it might get confusing.

**Comment by howardbaik** [comment]
@cansavvy thanks for your comments, I will merge this PR!

**Pull request merged**